crates/pop-contracts/src/new.rs

```diff
@@ -1,30 +1,51 @@
 use contract_build::new_contract_project;
 use std::path::Path;
+use thiserror::Error;
 
-/// Create a new smart contract at `target`
-pub fn create_smart_contract(name: String, target: &Path) -> anyhow::Result<()> {
-	// In this code, out_dir will automatically join `name` to `target`,
-	// which is created prior to the call to this function
-	// So we must pass `target.parent()`
-	new_contract_project(&name, target.canonicalize()?.parent())
+#[derive(Error, Debug)]
+pub enum Error {
+	#[error("Failed to create new contract project: {0}")]
+	NewContractError(String),
```

Comment:
I suggest to omit the `Error` suffix. When you're matching onn the error:
```rs
match fallible_operation() {
  Ok(_) => {}
  Err(e) => match e {
    Error::New,
    Error::Io,
    }
 ```
 helps eliminate some of the redundant visual noise.